trunner/harness/psh.py

```diff
@@ -81,4 +90,14 @@ def __call__(self, result: TestResult) -> TestResult:
                 ) from e
 
         result.set_stage(TestStage.RUN)
-        return self.next_harness(result)
+
+        if self.suppress_dmesg:
+            self.dut.pexpect_proc.sendline("dmesg -D")
+            self.dut.expect_exact(self.prompt)
+            test_result = self.next_harness(result)
```

Comment:
IMHO we should call next_harness only once (split the dmesg disable /enable into separate IFs in respective places)